hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -303,14 +337,133 @@ private void doPutsAndDeletes(HTable hTable, List<Put> puts, List<Delete> delete
     hTable.flushCommits();
     puts.clear();
     deletes.clear();
+    sleepForTime(SLEEP_TIME_MILLISECONDS);
+  }
+
+  private static void sleepForTime(int sleepTimeMs) {
+    try {
+      Thread.sleep(sleepTimeMs);
+    } catch (InterruptedException e) {
+      logger.error(e);
+      throw new RuntimeException(e);
+    }
   }
 
   @Override
   public JavaRDD<WriteStatus> updateLocation(JavaRDD<WriteStatus> writeStatusRDD, JavaSparkContext jsc,
       HoodieTable<T> hoodieTable) {
+    setPutBatchSize(writeStatusRDD, jsc);
     return writeStatusRDD.mapPartitionsWithIndex(updateLocationFunction(), true);
   }
 
+  private void setPutBatchSize(JavaRDD<WriteStatus> writeStatusRDD,
+      final JavaSparkContext jsc) {
+    if (config.getHbaseIndexPutBatchSizeAutoCompute()) {
+      SparkConf conf = jsc.getConf();
+      int maxExecutors = conf.getInt(DEFAULT_SPARK_EXECUTOR_INSTANCES_CONFIG_NAME, 1);
+      if (conf.getBoolean(DEFAULT_SPARK_DYNAMIC_ALLOCATION_ENABLED_CONFIG_NAME, false)) {
+        maxExecutors = Math.max(maxExecutors, conf.getInt(
+            DEFAULT_SPARK_DYNAMIC_ALLOCATION_MAX_EXECUTORS_CONFIG_NAME, 1));
+      }
+
+      /*
+        Each writeStatus represents status information from a write done in one of the IOHandles.
+        If a writeStatus has any insert, it implies that the corresponding task contacts HBase for
+        doing puts, since we only do puts for inserts from HBaseIndex.
+        Note that writeStatusRDD is cached in HoodieWriteClient before the control comes here.
```

Comment:
Let's not comment about other implementation like caching etc here, people should be able to figure this out based on usage.